FWCore/ParameterSet/python/Mixins.py

```diff
@@ -636,11 +636,18 @@ def _modifyParametersFromDict(params, newParams, errorRaiser, keyDepth=""):
                     if isinstance(params[key],_Parameterizable):
                         pset = params[key]
                         p =pset.parameters_()
+                        oldkeys = set(p.keys())
                         _modifyParametersFromDict(p,
                                                   value,errorRaiser,
                                                   ("%s.%s" if type(key)==str else "%s[%s]")%(keyDepth,key))
                         for k,v in p.iteritems():
                             setattr(pset,k,v)
+                            try:
+                                oldkeys.remove(k)
```

Comment:
If you switch to `oldkeys.discard(k)` you do not have to use the `try...except`

Reply:
Thanks, much cleaner with that.